ipfs-8-to-9/migration/migration.go

```diff
@@ -0,0 +1,110 @@
+// package mg8 contains the code to perform 8-9 repository migration in
+// go-ipfs. This performs a switch to raw multihashes for all keys in the
+// go-ipfs datastore (https://github.com/ipfs/go-ipfs/issues/6815).
+package mg8
+
+import (
+	"errors"
+	"fmt"
+
+	migrate "github.com/ipfs/fs-repo-migrations/go-migrate"
+	lock "github.com/ipfs/fs-repo-migrations/ipfs-1-to-2/repolock"
+	"github.com/ipfs/go-datastore/namespace"
+
+	mfsr "github.com/ipfs/fs-repo-migrations/mfsr"
+	log "github.com/ipfs/fs-repo-migrations/stump"
+	ds "github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-ipfs/plugin/loader"
+	fsrepo "github.com/ipfs/go-ipfs/repo/fsrepo"
+)
+
+// Migration implements the migration described above.
+type Migration struct{}
+
+// Versions returns the current version string for this migration.
+func (m Migration) Versions() string {
+	return "8-to-9"
+}
+
+// Reversible returns false. This migration cannot be reverted, as we do not
+// know which raw hashes were actually CIDv1s. However, things should work all
+// the same as they will be treated as CIDv0s in old versions anyways.
+func (m Migration) Reversible() bool {
+	return false
```

Comment:
Options:

* A: Say this migration is not reversible.
* B: Say this migration is reversible but treat it as no-op because things pretty much work as before if the user was to run with the previous go-ipfs version.
* C: Write down the list of all CIDs that have been migrated and re-use that to revert the migration.

I vote B unless I'm missing something.

Reply:
Yeah, C it is.